### PR TITLE
feat: 2.0.0 version

### PR DESCRIPTION
**Pull request by pbajaj-crest**
BREAKING CHANGE: depreciation of preserve infra functionality

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -80,11 +80,11 @@ outputs:
     description: 'Name of workflow triggered'
 runs:
   using: 'docker'
-  image: 'docker://ghcr.io/splunk/wfe-test-runner-action/wfe-test-runner-action:v1.6.9'
+  image: 'Dockerfile'
```
[the remark this answers is not in the document]
Comment (pbajaj-crest) [reply action.yaml:83]: @kdoroszko-splunk  to take the docker image from the local else it was taking from the v1.6.8

File: action.yaml [hunk 2]
```diff
@@ -43,8 +43,8 @@ inputs:
     description: 'Namespace where the workflow template lies'
     required: false
     default: 'default'
-  delay-destroy:
-    description: 'Delay destruction of the setup by 1h'
+  run-destroy:
```
[the remark this answers is not in the document]
Comment (pbajaj-crest) [reply action.yaml:46]: @kdoroszko-splunk  yes we need that input.

File: action.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kdoroszko-splunk) [reply action.yaml:83]: Is there a reason to take the docker image from the local instead of v1.6.9 repo?

[the remark this answers is not in the document]
Comment (pbajaj-crest) [reply action.yaml:83]: @kdoroszko-splunk  For the current changes in branch [branch](https://github.com/splunk/wfe-test-runner-action/tree/fix/ADDON-69999-deprecate-preserve-infra) i have take the local image of docker once that is merged with main we have to update that image with new version of docker image.

File: action.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kdoroszko-splunk) [reply action.yaml:46]: What about default value?

File: action.yaml [hunk 3]
```diff
@@ -43,10 +43,6 @@ inputs:
     description: 'Namespace where the workflow template lies'
     required: false
     default: 'default'
-  delay-destroy:
-    description: 'Delay destruction of the setup by 1h'
-    required: True
-    default: 'No'
```
Comment (mbruzda-splunk) [inline action.yaml:49]: If this is bool value, can we please use _True_ or _False_ here?

Comment (pbajaj-crest) [reply action.yaml:49]: @mbruzda-splunk, @kdoroszko-splunk  just  udated the snippet here and test the pipeline. its working without run_delay parameter

File: action.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbruzda-splunk) [reply action.yaml]: @pbajaj-crest

[the remark this answers is not in the document]
Comment (pbajaj-crest) [reply action.yaml]: @mbruzda-splunk  @kdoroszko-splunk  i have removed that parameter  and tested that. its working fine in [add-on](https://github.com/splunk/splunk-add-on-for-microsoft-cloud-services/actions/runs/8984472376/job/24676656038)

**Comment by mbruzda-splunk** [comment]
### Description

Deprecates preserve infra (delay-destroy) functionality.

1. Update the reusable code as per the requirement ([jira-ticket](https://splunk.atlassian.net/browse/ADDON-69999))
2. update the `delay_destroy` functionality


#### Related PRs:
https://github.com/splunk/addonfactory-workflow-addon-release/pull/262
https://github.com/splunk/ta-automation-k8s-manifests/pull/86
https://github.com/splunk/ta-automation-app-of-apps/pull/5

**Review by kdoroszko-splunk — APPROVED** [review]

**Pull request merged**